src/transformers/portable-text-transformer/portable-text-transformer.ts

```diff
@@ -60,322 +69,290 @@ import {
+  depth = 0,
+  lastListElement?: DomHtmlNode,
+  listType?: PortableTextListItemType,
+): PortableTextItem[] =>
+  nodes.flatMap((node: DomNode): PortableTextItem[] => {
+    let currentListType = listType;
+
+    if (isElement(node)) {
+      if (node.tagName === "td") {
+        return transformTableCell(node);
+      }
+
+      if (isListBlock(node)) {
+        lastListElement = node;
+        currentListType = isUnorderedListBlock(node) ? "bullet" : "number";
+      }
+
+      if (isListItem(node)) {
+        // set depth to 1 for the first list item, increment for each nested list
+        if (lastListElement && isListBlock(lastListElement)) {
+          depth++;
         }
+        // ensures depth remains the same until a nested listBlock is found
+        lastListElement = undefined;
+      }
 
-        // If not an element, transform as text and return as array
-        return [transformText(node)]
-    });
-};
+      // Recursively flatten children and concatenate with the transformed node.
+      const transformedNode = transformNode(node, depth, currentListType);
+      const transformedChildren = flatten(node.children, depth, lastListElement, currentListType);
 
-const transformNode = (node: DomNode, depth: number, listType?: PortableTextListItemType): PortableTextItem[] => {
-    if (isText(node)) {
-        return [transformText(node)];
-    } else {
-        return transformElement(node, depth, listType);
+      return [...transformedNode, ...transformedChildren];
     }
-}
 
-const transformElement = (node: DomHtmlNode, depth: number, listType?: PortableTextListItemType): PortableTextItem[] => {
-    const transformFunction = transformMap[node.tagName as ValidElement];
+    return [transformText(node)];
+  });
 
-    return transformFunction(node, depth, listType!);
-}
+const transformNode = (node: DomNode, depth: number, listType?: PortableTextListItemType): PortableTextItem[] =>
+  isText(node) ? [transformText(node)] : transformElement(node, depth, listType);
 
-const transformImage: TransformElementFunction<FigureElementAttributes> = (node) => {
-    const imageTag = node.children[0] as DomHtmlNode<ImgElementAttributes>;
-    const block = createImageBlock(uid().toString());
+const transformElement = (
+  node: DomHtmlNode,
+  depth: number,
+  listType?: PortableTextListItemType,
+): PortableTextItem[] => {
+  const transformFunction = transformMap[node.tagName as ValidElement];
 
-    block.asset._ref = node.attributes['data-asset-id'];
-    block.asset.url = imageTag.attributes['src'];
-    block.asset.alt = imageTag.attributes['alt'];
+  return listType !== undefined
+    ? (transformFunction as TransformListItemFunction)(node, depth, listType)
+    : (transformFunction as TransformElementFunction)(node);
+};
 
-    return [block];
-}
+const transformImage: TransformElementFunction<FigureElementAttributes> = (node) => {
+  const imageTag = node.children[0] as DomHtmlNode<ImgElementAttributes> | undefined;
```

Comment:
this is still not correct, it would be better to use some kind of typeguard or conditions at least that the `tagName` really is in the image tag, even though this code will actually behave kind of correctly, it's still kind of feels a workaround :/